libass/ass_render.c

```diff
@@ -1593,16 +1591,24 @@ wrap_lines_smart(ASS_Renderer *render_priv, double max_text_width)
     int i;
     GlyphInfo *cur, *s1, *e1, *s2, *s3;
     int last_space;
-    int last_cjk;
+    int last_break;
     int break_type;
     int exit;
     double pen_shift_x;
     double pen_shift_y;
     int cur_line;
     TextInfo *text_info = &render_priv->text_info;
+    char brks[text_info->length];
+    char *brksPtr;
+
+    if (render_priv->settings.wrap_tr14) {
+        brksPtr = brks;
+        set_linebreaks(text_info, text_info->length, "ja", brksPtr,
```

Comment:
Why the `brksPtr`? Why not just pass `brks`?

Reply:
Fixed. I also noticed that `lang` is hard-coded to `"ja"`. I'll have to investigate if it's possible to use the subtitle language and fall back to something based on the code points in the string.

Reply:
Good point. I think we already have a language field somewhere.